agentic-tutorial/src/main/java/_9_human_in_the_loop/_9b_HumanInTheLoop_Chatbot_With_Memory.java

```diff
@@ -0,0 +1,92 @@
+package _9_human_in_the_loop;
+
+import dev.langchain4j.agentic.AgenticServices;
+import dev.langchain4j.agentic.UntypedAgent;
+import dev.langchain4j.agentic.workflow.HumanInTheLoop;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+import dev.langchain4j.model.chat.ChatModel;
+import dev.langchain4j.service.AiServices;
+import util.ChatModelProvider;
+import util.log.CustomLogging;
+import util.log.LogLevels;
+
+import java.util.Map;
+
+public class _9b_HumanInTheLoop_Chatbot_With_Memory {
+
+    static {
+        CustomLogging.setLevel(LogLevels.PRETTY, 300);  // control how much you see from the model calls
+    }
+
+    /**
+     * This example demonstrates a back-and-forth loop with human-in-the-loop interaction,
+     * until an end-goal is reached (exit condition), after which the rest of the workflow
+     * can continue.
+     * The loop continues until the human confirms availability, which is verified by an AiService.
+     * When no slot is found, the loop ends after 5 iterations.
+     */
+
+    private static final ChatModel CHAT_MODEL = ChatModelProvider.createChatModel();
+
+    public static void main(String[] args) {
+
+        // 1. Define sub-agent
+        MeetingProposer proposer = AgenticServices
+                .agentBuilder(MeetingProposer.class)
+                .chatModel(CHAT_MODEL)
+                .chatMemoryProvider(memoryId -> MessageWindowChatMemory.withMaxMessages(15)) // so the agent remembers what he proposed already
+                .outputName("proposal")
+                .build();
+
+        // 2. Add an AiService to judge if a decision has been reached (this can be a tiny local model because the assignment is so simple)
+        DecisionsReachedService decisionService = AiServices.create(DecisionsReachedService.class, CHAT_MODEL);
+
+        // 2. Define Human-in-the-loop agent
+        HumanInTheLoop humanInTheLoop = AgenticServices
```

Comment:
Would it make sense to also have an example demonstrating the HITL with the `async` option?